gtdynamics.i

```diff
@@ -86,16 +86,9 @@ gtsam::NonlinearFactor TwistFactor(const gtsam::noiseModel::Base *cost_model,
                                    size_t t = 0);
 
 #include <gtdynamics/factors/TwistAccelFactor.h>
-class TwistAccelFactor : gtsam::NonlinearFactor {
-  TwistAccelFactor(gtsam::Key twist_key_c, gtsam::Key twistAccel_key_p,
-                   gtsam::Key twistAccel_key_c, gtsam::Key q_key,
-                   gtsam::Key qVel_key, gtsam::Key qAccel_key,
-                   const gtsam::noiseModel::Base *cost_model,
-                   const gtdynamics::Joint *joint);
-
-  void print(const string &s="",
-             const gtsam::KeyFormatter &keyFormatter=gtdynamics::GTDKeyFormatter);
-};
+gtsam::NonlinearFactor TwistAccelFactor(const gtsam::noiseModel::Base *cost_model,
```

Comment:
@dellaert this won't work. `TwistAccelFactor` returns an `ExpressionFactor` which is a type that is not wrapped. The moment you invoke the function from Python (or any wrapped function), the interpreter will crap out with an unknown type error.

Getting ExpressionFactor wrapped is very hard because it requires us to wrap Expression in general and that requires a lot of work with functionals. I don't know if you already looked into the feasibility of this from a wrapping perspective when you decided to make these changes.

Reply:
That makes sense. I'll try to make it return a boost::shared_ptr instead

Reply:
make the functions return NonlinearFactor::shared_ptr. In this way, we won't expose the ExpressionFactor to the Python wrapper, and the wrapper works just fine.

Reply:
That won't work either because a NonlinearFactor is an abstract base type. You need to return the actual type of the return value.

Reply:
@yetongumich I recommend adding a Python unit test rather than shooting arrows in the dark. This will ensure your changes are ironclad.

Reply:
I actually think this will work. As long it is a nonlinear factor shared pointer. But I don’t understand why we need them. I’d rather remove these factors from the wrapper. They are on the way out anyway.

@varunagrawal , you can use the temporary factor in your other PR.

Reply:
But in general I agree with @varunagrawal  comment, @yetongumich : have unit tests.

Reply:
Yetong mentioned using a NonlinearFactor but he actually is returning a NoiseModelFactor. That works because it is not an abstract base type, so his comment is misleading aka not cool bro.

@dellaert what do you mean by these factors are on their way out? Can you please elaborate on this since this was not mentioned to the rest of the group. How would this impact being able to construct specific factor graphs for various problems (and not just state estimation)?

Reply:
@varunagrawal Yes. I was initially returning NonlinearFactor::shared_ptr, but I later discovered that some unit tests require some methods in NoiseModelFactor, so I returned NoiseModelFactor::shared_ptr instead. They python wrapper works, but I'll add the unit tests as well.

Reply:
What do you mean by the python wrapper works? That it compiles successfully? That is not the same as it working.

Reply:
@varunagrawal please meet me on Bluejeans now if you can.
@yetongumich please don't add any unit tests for now until I understand Varun's objection, after he has the whole context.